doc/radosgw/rgw-cache.rst

```diff
@@ -0,0 +1,98 @@
+========================== 
+RGW Data caching and CDN
+==========================
+
+.. versionadded:: Octopus
+
+.. contents::
+
+This new feature adds to RGW the ability to securely cache objects and offload the workload from the cluster, using Nginx.
+After an object is accessed the first time it will be stored on top of Nginx dir.
+Every request the Nginx will not fetch data from the RGW or the cluster, It will only check for permissions for that user from the RGW.
+This feature is based on some Nginx modules, ngx_http_auth_request_module, https://github.com/kaltura/nginx-aws-auth-module, Openresty for lua capablities.     
+Currently this feature only works for GET requests and it will cache only AWSv4 requests (only s3 requests).
+The feature introduces 2 new APIs: Auth and Cache.
+
+New APIs
+-------------------------
+
+There are 2 new apis for this feature:
+
+Auth API - Nginx using it to validate that an user can access the cached data
+
+Cache API - Adding the ability to override securely Range header, that way Nginx can use it is own smart cache on top of S3:
+https://www.nginx.com/blog/smart-efficient-byte-range-caching-nginx/
+Using this API giving the ability to read ahead objects when clients asking a specific range from the object. 
+The second time the client will ask another range, the Nginx will have it and it will provide the other range from the cache.
```

Comment:
On subsequent accesses to the cached object, Nginx will satisfy requests for already-cached ranges from cache.  Uncached ranges will be read from RGW (and cached).